integration/mock-backend-server/src/main/java/org/wso2/choreo/connect/mockbackend/MockBackendSOAP.java

```diff
@@ -0,0 +1,119 @@
+import static com.github.tomakehurst.wiremock.client.WireMock.equalToXml;
+import static com.github.tomakehurst.wiremock.client.WireMock.containing;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
+
+public class MockBackendSOAP extends Thread{
+    private static final org.slf4j.Logger log = LoggerFactory.getLogger(MockBackendSOAP.class);
+    private WireMockServer wireMockServer;
+    private String wsdlDefinition;
+    private String responseBodySoap11;
+    private String responseBodySoap12;
+
+    private static final String REQ_PAYLOAD_11 = "<soap:Envelope\n" +
+            "\txmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\n" +
+            "\txmlns:xsd=\"http://www.w3.org/2001/XMLSchema\"\n" +
+            "\txmlns:soap=\"http://schemas.xmlsoap.org/soap/envelope/\">\n" +
+            "\t<soap:Body>\n" +
+            "\t\t<CheckPhoneNumber\n" +
+            "\t\t\txmlns=\"http://ws.cdyne.com/PhoneVerify/query\">\n" +
+            "\t\t\t<PhoneNumber>18006785432</PhoneNumber>\n" +
+            "\t\t\t<LicenseKey>18006785432</LicenseKey>\n" +
+            "\t\t</CheckPhoneNumber>\n" +
+            "\t</soap:Body>\n" +
+            "</soap:Envelope>";
+
+    private static final String REQ_PAYLOAD_12 = "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n" +
+            "<soap12:Envelope\n" +
+            "\txmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\n" +
+            "\txmlns:xsd=\"http://www.w3.org/2001/XMLSchema\"\n" +
+            "\txmlns:soap12=\"http://www.w3.org/2003/05/soap-envelope\">\n" +
+            "\t<soap12:Body>\n" +
+            "\t\t<CheckPhoneNumber\n" +
+            "\t\t\txmlns=\"http://ws.cdyne.com/PhoneVerify/query\">\n" +
+            "\t\t\t<PhoneNumber>18006785432</PhoneNumber>\n" +
+            "\t\t\t<LicenseKey>18006785432</LicenseKey>\n" +
+            "\t\t</CheckPhoneNumber>\n" +
+            "\t</soap12:Body>\n" +
+            "</soap12:Envelope>";
+
+    public MockBackendSOAP(){
+        ClassLoader classloader = Thread.currentThread().getContextClassLoader();
+        try {
+            wsdlDefinition = Utils.readFileFromInputStream(
+                    classloader.getResourceAsStream("wsdl/PhoneVerification.wsdl"));
+            responseBodySoap11 = Utils.readFileFromInputStream(
+                    classloader.getResourceAsStream("soap/checkPhoneNumberResponseBody_11.xml"));
+            responseBodySoap12 = Utils.readFileFromInputStream(
+                    classloader.getResourceAsStream("soap/checkPhoneNumberResponseBody_12.xml"));
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void run(){
+        startWiremockServer(Constants.MOCK_BACKEND_SOAP_SERVER_PORT);
+    }
+
+    private void startWiremockServer(int port) {
+        log.info("Starting mock backend for SOAP service...");
```

Comment:
Let's also log the port number.

Reply:
fixed in https://github.com/wso2/product-microgateway/pull/2972/commits/5c91048f9fa451e4b2f2d91dac704d3f6b755cd3